ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/AbstractIgnoringFutureHistoricalSlot.java

```diff
@@ -46,12 +48,13 @@ public AbstractIgnoringFutureHistoricalSlot(
     this.spec = spec;
     this.futureSlotTolerance = futureSlotTolerance;
     this.historicalSlotTolerance = historicalSlotTolerance;
+    this.slotsInEpoch = UInt64.valueOf(spec.getGenesisSpec().getSlotsPerEpoch());
   }
 
   @Override
   public void onSlot(final UInt64 slot) {
     currentSlot = slot;
-    if (currentSlot.mod(historicalSlotTolerance).equals(UInt64.ZERO)) {
+    if (currentSlot.mod(slotsInEpoch).equals(PRUNE_SLOT)) {
```

Comment:
I really do think that we should change `onSlot(slot)` and make it `onSlot(slot, epoch, slotInEpoch)` so we damn calculate this things once